src/main/java/appeng/core/features/registries/GrinderRecipeManager.java

```diff
@@ -274,26 +325,66 @@ private void addDust( final String name, final ItemStack item )
 		}
 	}
 
-	@Override
-	public void oreRegistered( final String name, final ItemStack item )
+	private void log( final String o, Object... params )
 	{
-		if( name.startsWith( "ore" ) || name.startsWith( "crystal" ) || name.startsWith( "gem" ) || name.startsWith( "ingot" ) || name.startsWith( "dust" ) )
+		AELog.grinder( o, params );
+	}
+
+	private static class CacheKey
+	{
+		private final Item item;
+		private final int damage;
+
+		public CacheKey( ItemStack input )
 		{
-			for( final String ore : AEConfig.instance.grinderOres )
+			Preconditions.checkNotNull( input );
+			Preconditions.checkNotNull( input.getItem() );
+
+			this.item = input.getItem();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Method new appeng.core.features.registries.GrinderRecipeManager$CacheKey(ItemStack) appears to call the same method on the same object redundantly [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=fb-contrib%3APRMC_POSSIBLY_REDUNDANT_METHOD_CALLS)